lib/awsapi/config.go

```diff
@@ -246,13 +254,20 @@ func (cfg *Config) getAPICredentials() (*credentials, error) {
 	return acNew, nil
 }
 
-// getECSRoleCredentialsByPath makes request to ecs metadata service
-// and retrieves instances credentials
+// getCredentialsByPath makes request to ECS and EKS metadata service
+// and retrieves instance and container credentials
 // https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html
-func getECSRoleCredentialsByPath(client *http.Client, path string) (*credentials, error) {
-	resp, err := client.Get(path)
+func getCredentialsByPath(client *http.Client, path, token string) (*credentials, error) {
+	req, err := http.NewRequest("GET", path, nil)
+	if err != nil {
+		return nil, err
+	}
+	if len(token) > 0 {
+		req.Header.Add("Authorization", token)
+	}
+	resp, err := client.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("cannot get ECS instance role credentials: %w", err)
+		return nil, fmt.Errorf("cannot get credentials by path: %w", err)
```

Comment:
does it make sense to print `path` in error message? Or it could contain sensetive info?

Reply:
The ECS path is not sensitve, and usually it is a constant value. Something like `http://169.254.170.2/xxx`.

Reply:
Path could be obtained from env variable and it cannot be constant after this change. It makes sense to add to the error context.

Reply:
added path to error message